cc.craftos_pc.craftos_pc.yaml

```diff
@@ -0,0 +1,121 @@
+    buildsystem: cmake-ninja
+    sources:
+      - type: git
+        url: https://github.com/libharu/libharu.git
+        tag: v2.4.3
+        commit: 8dbcfe470581edc4bc52c3ef132d410e66cadc20
+  
+  - name: pngpp
+    buildsystem: simple
+    build-commands:
+      - make install PREFIX=/app
+    sources:
+      - type: archive
+        url: http://download.savannah.nongnu.org/releases/pngpp/png++-0.2.10.tar.gz
+        sha512: c54a74c0c20212bd0dcf86386c0c11dd824ad14952917ba0ff4c184b6547744458728a4f06018371acb7d5b842b641708914ccaa81bad8e72e173903f494ca85
+  
+  - shared-modules/SDL2/SDL2-with-libdecor.json
+  
+  - name: patchelf
+    buildsystem: autotools
+    cleanup:
+      - "*"
+    sources:
+      - type: git
+        url: https://github.com/NixOS/patchelf.git
+        tag: "0.17.2"
+        commit: 5908e16cd562bcb1909be4de0409c4912a8afc52
+  
+  - name: craftos2
+    buildsystem: simple
+    build-commands: 
+      - make -C craftos2-lua -j$FLATPAK_BUILDER_N_JOBS linux
+      - ./configure --without-ncurses --prefix=/app
+      - make -j$FLATPAK_BUILDER_N_JOBS
+
+      # https://github.com/MCJack123/craftos2/issues/237
+      - install -D -m 755 craftos2-lua/src/liblua.so /app/lib/libcraftos2-lua.so
+      - patchelf --replace-needed craftos2-lua/src/liblua.so libcraftos2-lua.so craftos
+
+      - install -D -m 755 craftos /app/bin/craftos
+
+      - mkdir icons
+      - unzip resources/linux-icons.zip -d icons
+      - install -D -m 0644 icons/CraftOS-PC.desktop "/app/share/applications/cc.craftos_pc.craftos_pc.desktop"
+      - desktop-file-edit --set-icon="cc.craftos_pc.craftos_pc" --set-key="Exec" --set-value="craftos" /app/share/applications/cc.craftos_pc.craftos_pc.desktop
+      - install -D -m 0644 icons/16.png "/app/share/icons/hicolor/16x16/apps/cc.craftos_pc.craftos_pc.png"
+      - install -D -m 0644 icons/24.png "/app/share/icons/hicolor/24x24/apps/cc.craftos_pc.craftos_pc.png"
+      - install -D -m 0644 icons/32.png "/app/share/icons/hicolor/32x32/apps/cc.craftos_pc.craftos_pc.png"
+      - install -D -m 0644 icons/48.png "/app/share/icons/hicolor/48x48/apps/cc.craftos_pc.craftos_pc.png"
+      - install -D -m 0644 icons/96.png "/app/share/icons/hicolor/96x96/apps/cc.craftos_pc.craftos_pc.png"
+      - install -D -m 0644 128.png "/app/share/icons/hicolor/128x128/apps/cc.craftos_pc.craftos_pc.png"
+
+      - install -D -m 0644 cc.craftos_pc.craftos_pc.appdata.xml /app/share/appdata/cc.craftos_pc.craftos_pc.appdata.xml
+
+    sources:
+      - type: git
+        url: https://github.com/MCJack123/craftos2.git
+        tag: v2.7.2
+        commit: e4ed6c0527d9983154df9b67d26593f84a4f2edc
+        x-checker-data:
+          type: git
+          tag-pattern: "^v([\\d.]+)$"
+      - type: file
+        path: cc.craftos_pc.craftos_pc.appdata.xml
+      # upstream does not contain a 128 sized icon to pass flathub build, here is one downscaled from an upstream 1024x1024 icon
+      - type: file
+        path: icons/128.png
+    
```

Comment:
It would be better to downscale it as part of the build to make sure it's up to date, and if there's no SVG, 512x512 is preferred.